crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -93,6 +93,33 @@ use std::{
 ///
 /// This will most commonly occur when working with `SystemParam`s generically, as the requirement
 /// has not been proven to the compiler.
+///
+/// # `!Sync` Resources
+/// A `!Sync` type cannot implement `Resource`. However, it is possible to wrap a `Send` but not `Sync`
+/// type in [`SyncCell`] or the currently unstable [`Exclusive`] to make it `Sync`. This forces only
+/// having mutable access (`&mut T` only, never `&T`), but makes it safe to reference across multiple
+/// threads.
+///
+/// ```compile_fail
+/// # use std::cell::RefCell;
+/// # use bevy_ecs::system::Resource;
+/// use bevy_utils::synccell::SyncCell;
+///
+/// // This will fail to compile since RefCell is !Sync.
+/// #[derive(Resource)]
+/// struct NotSync {
+///    counter: RefCell<usize>,
+/// }
+///
```

Comment:
Could tighten the `compile_fail` test while ensuring the part that should compile compiles, and rendering the same in docs

```suggestion
/// ```
///
/// ```
/// # use std::cell::RefCell;
/// # use bevy_ecs::system::Resource;
/// # use bevy_utils::synccell::SyncCell;
```